code/game/machinery/droneDispenser.dm

```diff
@@ -126,132 +155,179 @@
-		if(stack <= 0)
-			return
-		if(!user.canUseTopic(src))
+			user << "<span class='warning'>There isn't a place \
+				to insert [O]!</span>"
 			return
+		var/obj/item/stack/sheets = O
 		if(!user.canUnEquip(sheets))
-			user << "<span class='warning'>[O] is stuck to your hand, you can't get it off!</span>"
+			user << "<span class='warning'>[O] is stuck to your hand, \
+				you can't get it off!</span>"
 			return
-		stack = Clamp(stack, 1, sheets.max_amount)
-		stack = materials.insert_stack(sheets, stack)
-		if(stack)
-			user << "<span class='notice'>You insert [stack] sheet[stack > 1 ? "s" : ""] into [src].</span>"
+
+		var/used = materials.insert_stack(sheets, sheets.amount)
+
+		if(used)
+			user << "<span class='notice'>You insert [used] \
+				sheet[used > 1 ? "s" : ""] into [src].</span>"
 		else
-			user << "<span class='warning'>The [src] can't use [sheets].</span>"
+			user << "<span class='warning'>The [src] isn't accepting the \
+				[sheets].</span>"
 
 	else if(istype(O, /obj/item/weapon/weldingtool))
-		if(stat & BROKEN)
-			var/obj/item/weapon/weldingtool/WT = O
-			if(!WT.isOn())
-				return
-			if(WT.get_fuel() < 1)
-				user << "<span class='warning'>You need more fuel to complete this task!</span>"
-				return
-			playsound(src, 'sound/items/Welder.ogg', 50, 1)
-			user.visible_message("<span class='notice'>[user] begins patching up [src] with [WT].</span>", \
-								 "<span class='notice'>You begin restoring the damage to [src]...</span>")
-			if(!do_after(user, 40/O.toolspeed, target = src))
-				return
-			if(!src || !WT.remove_fuel(1, user)) return
-			user.visible_message("<span class='notice'>[user] fixes [src]!</span>", \
-								 "<span class='notice'>You restore [src] to operation.</span>")
-			stat &= ~BROKEN
-			health = max_health
-			if(!stat)
-				icon_state = icon_on
-		else
+		if(!(stat & BROKEN))
 			user << "<span class='warning'>[src] doesn't need repairs.</span>"
+			return
+
+		var/obj/item/weapon/weldingtool/WT = O
+
+		if(!WT.isOn())
+			return
+
+		if(WT.get_fuel() < 1)
+			user << "<span class='warning'>You need more fuel to \
+				complete this task!</span>"
+			return
+
+		playsound(src, 'sound/items/Welder.ogg', 50, 1)
+		user.visible_message(
+			"<span class='notice'>[user] begins patching up \
+				[src] with [WT].</span>",
+			"<span class='notice'>You begin restoring the \
+				damage to [src]...</span>")
+
+		if(!do_after(user, 40/O.toolspeed, target = src))
+			return
+		if(!src || !WT.remove_fuel(1, user))
+			return
+
+		user.visible_message(
+			"<span class='notice'>[user] fixes [src]!</span>",
+			"<span class='notice'>You restore [src] to operation.</span>")
+
+		stat &= ~BROKEN
+		health = max_health
+		update_icon()
 	else
 		return ..()
 
-/obj/machinery/droneDispenser/take_damage(damage, damage_type = BRUTE, sound_effect = 1)
-	switch(damage_type)
+/obj/machinery/droneDispenser/take_damage(dmg, dmg_type=BRUTE, hit_sound=TRUE)
```

Comment:
why change the argument names? I standardized those over all machines and structures.